src/libs/vehicle/mavlink/vehicle.ts

```diff
@@ -773,6 +774,68 @@ export abstract class MAVLinkVehicle<Modes> extends Vehicle.AbstractVehicle<Mode
     sendMavlinkMessage(heartbeatMessage)
   }
 
+  /**
+   * Send system time from GCS
+   */
+  sendSystemTime(): void {
+    const systemTimeMessage: Message.SystemTime = {
+      type: MAVLinkType.SYSTEM_TIME,
+      time_unix_usec: Math.floor(performance.now() * 1000),
+      time_boot_ms: 0,
```

Comment:
```suggestion
    const time_since_gcs_start_ms = performance.now()
    const systemTimeMessage: Message.SystemTime = {
      type: MAVLinkType.SYSTEM_TIME,
      time_unix_usec: Math.floor(performance.timeOrigin + time_since_gcs_start_ms) * 1000),
      time_boot_ms: Math.floor(time_since_gcs_start_ms),
```
Since this might be used to set the timestamps of log files and the like, the unix timestamp should be valid.